src/plugins/intel_gna/src/transformations/aszp_decomposition.cpp

```diff
@@ -0,0 +1,231 @@
+	uint32_t pad_begin,
+	uint32_t pad_end,
+	std::shared_ptr<ov::Node> padding_const,
+	std::shared_ptr<ov::Node> input_node) {
+
+	OutputVector concat_vector;
+	if (pad_begin > pad_end) {
+		concat_vector.push_back(padding_const->output(0));
+		concat_vector.push_back(input_node->output(0));
+	}else {
+		concat_vector.push_back(input_node->output(0));
+		concat_vector.push_back(padding_const->output(0));	
+	}
+	return std::make_shared<ngraph::op::Concat>(concat_vector,1);
+
+}
+
+
+
+
+Output<Node> DecomposeHeight(Output<Node> input, ov::CoordinateDiff pads_begin, ov::CoordinateDiff pads_end, ov::Shape conv_input_shape) {
+	uint32_t Hbegin, Hend, Wbegin, Wend, Hpad, Wpad;
+	uint64_t N, C, H, W;
+    std::tie(Hbegin, Hend, Wbegin, Wend, Hpad, Wpad) = ExtractPadding(pads_begin, pads_end);
+    std::tie(N,H,W,C) = GetInputDimensions(conv_input_shape);
+
+	if (Hpad != 0) {
+		auto new_reshape   = CreateReshape(input, 2, Shape{H,W*C});
+		auto new_transpose = CreateTranspose(new_reshape->output(0));
+
+		auto padding_const = CreateZeroConst(Shape{W*C,Hpad});
+		auto new_concat    = ConcatZeros(Hbegin,Hend,padding_const,new_transpose);
+		
+		auto new_untranspose = CreateTranspose(new_concat->output(0));	// Shape {W*C,H+Hpad}
+
+		if (0 == Wpad) return CreateReshape(new_untranspose->output(0), 4, Shape{N, H+Hpad, W ,C})->output(0);
+		else return (new_untranspose->output(0));
+	}else return input;
+
+
+}
+
+
+
+Output<Node> DecomposeWidth(Output<Node> input, ov::CoordinateDiff pads_begin, ov::CoordinateDiff pads_end, ov::Shape conv_input_shape) {
+	uint32_t Hbegin, Hend, Wbegin, Wend, Hpad, Wpad;
+	uint64_t N, H, W, C;
+    std::tie(Hbegin, Hend, Wbegin, Wend, Hpad, Wpad) = ExtractPadding(pads_begin, pads_end); //todo cleanup. maybe get only what we need?
+    std::tie(N,H,W,C) = GetInputDimensions(conv_input_shape);
+
+	if (Wpad != 0) {
+		auto new_reshape   = CreateReshape(input, 2, Shape{(H+Hpad)*W, C});
+		auto new_transpose = CreateTranspose(new_reshape->output(0));
+		auto new_reshape2  = CreateReshape(new_transpose->output(0),2, Shape{C*(H+Hpad), W});
+
+		auto padding_const = CreateZeroConst( Shape{C*(H+Hpad),Wpad} );
+		auto new_concat    = ConcatZeros(Wbegin,Wend,padding_const,new_reshape2);
+
+		auto new_unshape2    = CreateReshape(new_concat->output(0), 2, Shape{C, (H + Hpad) * (W + Wpad)});
+		auto new_untranspose = CreateTranspose(new_unshape2->output(0));
+		auto new_unshape     = CreateReshape(new_untranspose->output(0), 4, {N, H+Hpad, W+Wpad ,C});
+
+		return new_unshape->output(0);
+	}else return input;
+}
+
+
+
+void TrimmPadding(ov::CoordinateDiff& pads_begin, ov::CoordinateDiff& pads_end) {
+	if ( pads_begin[0] > pads_end[0]) 
+		 pads_begin[0] = pads_end[0];
+	else pads_end[0] = pads_begin[0];
+	if ( pads_begin[1] > pads_end[1]) 
+		 pads_begin[1] = pads_end[1];
+	else pads_end[1] = pads_begin[1];
+}
+
+
+std::shared_ptr<ov::Node> CreateConvolution (
+	std::shared_ptr<ov::Node> conv, 
```

Comment:
why do you pass node std::shared_ptr<ov::Node> conv for here if in place where you call it you already have 
std::shared_ptr<ngraph::opset8::Convolution> conv ?

Reply:
I guess I just followed the idea used in a different transform. So I just changed that. Thank you